### ISSUE TITLE
what is cz_mask？

### ISSUE THREAD

**Issue by horacehht** [OscarPellicer/prostate_lesion_detection#2]
Thank you for sharing such a complete task solution.And while reading your solution in ProstateX preprocessing.ipynb, I came up with following questions:
1. what is cz_mask? what does cz mean?(I have little domain knowledge in medical image)
2. how to produce cz_mask?(I know the Prostate mask can be generated by the software "3D Slicer" artificially)

I would be appreciated if u could answer my questions in your spare time.

**Comment by OscarPellicer** [comment]
Hi, thank you for your interest!
CZ means "Central Zone", but this name is not 100% correct, and it should be called central gland instead (as it is called in the paper: https://www.nature.com/articles/s41598-022-06730-6), because cz_masks actually contain both the central zone as well as the transitional zone of the prostate according to McNeal's prostate regions.
Central gland masks are produced automatically by using a CNN-based deep learning model that is not included in this repository (see the paper again for the details).

Regards,
Oscar

**Comment by horacehht** [comment]
> Hi, thank you for your interest! CZ means "Central Zone", but this name is not 100% correct, and it should be called central gland instead (as it is called in the paper: https://www.nature.com/articles/s41598-022-06730-6), because cz_masks actually contain both the central zone as well as the transitional zone of the prostate according to McNeal's prostate regions. Central gland masks are produced automatically by using a CNN-based deep learning model that is not included in this repository (see the paper again for the details).
> 
> Regards, Oscar

Thanks for your response in such a short time. Knowing the paper you list in the reply, I spent nearly two days reading it. 
I found the segmentation part at page 8 in the paper. You mentioned that "the first CNN model takes a prostate T2 image as input" and the following paragraph says"the CG was mannully segmented by a radiologist with two years of experience in PCa imaging". Does it mean that **the needed inputs for two CNNs** are Central Gland masks which are **mannully segmented by radiologists**? **Only need CG masks**?
I would be very grateful if u can solve my doubts.

**Comment by OscarPellicer** [comment]
There are two masks per prostate: whole prostate masks and central gland masks. Both of them are provided in `ProstateX_masks.zip` for all patients, so you don't need to worry about generating them yourself.

However, if you are interested in what I did to generate them, I trained two different CNN models specifically for that. The first one uses the model described [here](https://www.mdpi.com/2076-3417/11/2/844) to produce the prostate masks by using many manually-segmented examples from different datasets. The second model takes both the T2 MRI and the mask generated by the previous model and outputs the central gland mask, as described [here](https://www.nature.com/articles/s41598-022-06730-6). This second model is trained on a smaller dataset prepared specifically for this problem. By having these two additional models, you can automate the whole detection process, and you don't depend on humans having to segment the prostate for you beforehand.

**Comment by horacehht** [comment]
> There are two masks per prostate: whole prostate masks and central gland masks. Both of them are provided in `ProstateX_masks.zip` for all patients, so you don't need to worry about generating them yourself.
> 
> However, if you are interested in what I did to generate them, I trained two different CNN models specifically for that. The first one uses the model described [here](https://www.mdpi.com/2076-3417/11/2/844) to produce the prostate masks by using many manually-segmented examples from different datasets. The second model takes both the T2 MRI and the mask generated by the previous model and outputs the central gland mask, as described [here](https://www.nature.com/articles/s41598-022-06730-6). This second model is trained on a smaller dataset prepared specifically for this problem. By having these two additional models, you can automate the whole detection process, and you don't depend on humans having to segment the prostate for you beforehand.

So sorry to bother you.The reason why I asked the questions above is that my teacher got a MRI imaging project from hospital. I am working on this project, and **need to confirm with the hospital** what data they should give to accomplish the tasks.
So I want to ask about **which part of the prostate the radiologists should label**. As far as I know now, radiologists should label the prostate zone and central zone, right?(Because I saw the paper _Deep learning for fully automatic detection,  segmentation, and Gleason grade estimation of prostate cancer in multiparametric magnetic resonance images_  mentiond the first CNN "takes a prostate T2 image as input and produces a prostate segmentation mask as output" and second CNN"takes both T2 and the prostate segmentation obtained in the previous step")

**Comment by horacehht** [comment]
Additionally, I have read two papers written by your team carefully. The content is great!

**Comment by OscarPellicer** [comment]
Thanks! It is as you mention: they should segment whole prostate + central gland, or peripheral zone + central gland. Either option is fine since whole prostate = central gland + peripheral zone.

**Comment by horacehht** [comment]
Today I find cz_mask has **three values** when I used `np.unique(cz_mask)`, [0, 1, 2].Previously u say that radiologists should label central gland and central gland consists of central zone and the transitional zone.**Does numbers [0, 1, 2] corresponds to not ROI, central zone, transitional zone?**
In addition, u say that the model which generates masks is not in the repository. Does it mean that I need to build the model by myself according to your paper?(Iaugh)
I would appreciate it if you could reply me~

**Comment by OscarPellicer** [comment]
> Does numbers [0, 1, 2] corresponds to not ROI, central zone, transitional zone?

No:

- 0: No ROI, 
- 1: transition zone, 
- 2: central zone.

In any case, you should be able to figure that out by yourself by plotting the masks.

> Does it mean that I need to build the model by myself according to your paper?

No, it means that you have to provide the masks yourself. If you generate them manually, then there is no need for the segmentation models.

**Comment by horacehht** [comment]
Our team plans to build a full-automatic system for prostate lesion detection like your team. (segmentation first, then detection on the segmentation area)So it seems that we need to build a segmentation model as your paper _Robust Resolution-Enhanced Prostate Segmentation in Magnetic Resonance and Ultrasound Images through Convolutional Neural Networks_ says and we need to collect the whole prostate zone mask, central zone mask and transition zone mask for training.
Thank you very much for answering my mountains of questions.

**Comment by OscarPellicer** [comment]
In the case of a fully automatic system you do need to train those segmentation models. You do not need to use the specific segmentation architecture that I use though. There are many 3D Unet-like implementations in Github that you can use.

**Comment by horacehht** [comment]
> Thanks! It is as you mention: they should segment whole prostate + central gland, or peripheral zone + central gland. Either option is fine since whole prostate = central gland + peripheral zone.

I came up with a question. Should they segment the prostate + central gland on all the sequences?Or just segment on T2?
I would appreciate it if u could reply me.

**Comment by OscarPellicer** [comment]
In prostate MRI, T2 sequence is usually the reference, as it is where most structures can be seen. The rest of the sequences are then typically registered to the T2. So answering your question, they should probably refer everything to the T2. There is no need to segment more sequences, as the prostate is the same for all of them. The other should be registered to the T2 though.

**Comment by yinyin-llll** [comment]
> In prostate MRI, T2 sequence is usually the reference, as it is where most structures can be seen. The rest of the sequences are then typically registered to the T2. So answering your question, they should probably refer everything to the T2. There is no need to segment more sequences, as the prostate is the same for all of them. The other should be registered to the T2 though.

Friend, could you share your dataset? The official download channel for the dataset has been removed.

**Comment by yinyin-llll** [comment]
> Thank you for sharing such a complete task solution.And while reading your solution in ProstateX preprocessing.ipynb, I came up with following questions:
> 
> 1. what is cz_mask? what does cz mean?(I have little domain knowledge in medical image)
> 2. how to produce cz_mask?(I know the Prostate mask can be generated by the software "3D Slicer" artificially)
> 
> I would be appreciated if u could answer my questions in your spare time.

Friend, could you share your dataset? The official download channel for the dataset has been removed.
